core/modules/widgets/link.js

```diff
@@ -67,18 +67,22 @@ LinkWidget.prototype.renderLink = function(parent,nextSibling) {
 	if(this.linkClasses) {
 		classes.push(this.linkClasses);
 	}
-	classes.push("tc-tiddlylink");
-	if(this.isShadow) {
-		classes.push("tc-tiddlylink-shadow");
-	}
-	if(this.isMissing && !this.isShadow) {
-		classes.push("tc-tiddlylink-missing");
-	} else {
-		if(!this.isMissing) {
-			classes.push("tc-tiddlylink-resolves");
+	if(!this.enforceClass) {
+		classes.push("tc-tiddlylink");
+		if(this.isShadow) {
+			classes.push("tc-tiddlylink-shadow");
+		}
+		if(this.isMissing && !this.isShadow) {
+			classes.push("tc-tiddlylink-missing");
+		} else {
+			if(!this.isMissing) {
+				classes.push("tc-tiddlylink-resolves");
+			}
 		}
 	}
-	domNode.setAttribute("class",classes.join(" "));
+	if(classes.length) {
+		domNode.setAttribute("class",classes.join(" "));
+	}
```

Comment:
```
["asdf"].join("     ")   // returns "asdf"
["asdf"].join("     ").length  // returns 4
```

So no if needed. ... Or did I miss something?

Reply:
You did, I am validating the length of an array, not of a string.

Reply:
>You did, I am validating the length of an array, not of a string.

I know, what you did. ... I shouldn't have included my .length example, because it confused you. 

I see classes.length .. 0 is falsy, so you skip the "setAttribute"  so for me `classes.length > 0` would have been much clearer.

Reply:
@pmario: What is preferable about `classes.length > 0`? Is truthy not good enough?